torch/csrc/distributed/autograd/engine/dist_engine.cpp

```diff
@@ -210,6 +210,7 @@ std::shared_ptr<rpc::FutureMessage> DistEngine::runEngineAndAccumulateGradients(
               DistAutogradContainer::getInstance().getWorkerId(),
               ": ",
               error->what());
+          autogradContext->setAutogradError(errorMsg);
```

Comment:
Shouldn't we just call `propagateAutogradError` since this node has already received an exception, so there is no point setting an exception on the GraphTask again.

Also, isn't it sufficient to do this just in DistEngine::Execute? Since the node that drives distributed autograd, always receives an error and that node can propagate the error. In DistEngine::Execute() instead of just waiting on the future, you can add a callback to the future which propagates the error and then wait on the future.